src/main/java/org/folio/rest/impl/UserGroupAPI.java

```diff
@@ -40,323 +32,176 @@
+
+    // PgUtil.getById .. returns 500 where the below returns 404
+    Criterion c = new Criterion(
+      new Criteria().addField(ID_FIELD_NAME).setJSONB(false).setOperation("=").setValue("'" + groupId + "'"));
+
+    PostgresClientUtil.getInstance(vertxContext, okapiHeaders).get(GROUP_TABLE, Usergroup.class, c, false,
+      reply -> {
+        if (reply.succeeded()) {
+          @SuppressWarnings("unchecked")
+          List<Usergroup> userGroup = reply.result().getResults();
+          if (userGroup.isEmpty()) {
+            asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
+              .respond404WithTextPlain(groupId)));
+          } else {
+            asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
+              .respond200WithApplicationJson(userGroup.get(0))));
+          }
+        } else {
+          log.error(reply.cause().getMessage(), reply.cause());
+          if (isInvalidUUID(reply.cause().getMessage())) {
+            asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
+              .respond404WithTextPlain(groupId)));
+          } else {
+            asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(GetGroupsByGroupIdResponse
+              .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+          }
+        }
+      });
   }
 
   @Validate
   @Override
   public void deleteGroupsByGroupId(String groupId, String lang, Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-
-    vertxContext.runOnContext(v -> {
-      System.out.println("sending... deleteGroupsByGroupId");
-      try {
-        Criterion criterion = new Criterion(
-          new Criteria()
-            .addField(ID_FIELD_NAME)
-            .setJSONB(false)
-            .setOperation("=")
-            .setValue("'"+ groupId +"'"));
-        PostgresClient postgresClient = PostgresClientUtil.getInstance(vertxContext, okapiHeaders);
-        postgresClient.get(GROUP_TABLE, Usergroup.class, criterion, true, getReply -> {
-          try {
-            if(getReply.failed()) {
-               log.error(getReply.cause().getMessage(), getReply.cause());
-               asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(DeleteGroupsByGroupIdResponse
-                 .respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    Criterion criterion = new Criterion(
+      new Criteria()
+        .addField(ID_FIELD_NAME)
+        .setJSONB(false)
+        .setOperation("=")
+        .setValue("'" + groupId + "'"));
+    PostgresClient postgresClient = PostgresClientUtil.getInstance(vertxContext, okapiHeaders);
+    postgresClient.get(GROUP_TABLE, Usergroup.class, criterion, true, getReply -> {
```

Comment:
If this is really needed it should be replaced by PgUtil.getById

However, PgUtil.deleteById automatically returns 404 if the records doesn't exists. Therefore we may skip this check.

If we were using a foreign key relationship (schema.json) between users and user groups we can simply invoke deleteById and it will fail properly if the user group is still in use, no manual check needed.

Reply:
I'll need a commit from your to clarify. Or we skip this. Thanks.